### ISSUE TITLE
MA0076 should not be reported for invariant formats

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#491]
Similar to https://github.com/meziantou/Meziantou.Analyzer/issues/473, but now for **MA0076**.

Example:
```
var date = new DateOnly(2023, 4, 7);
var banner = $"Tomorrow is {date:o}";
```

**MA0076** will be reported for the last line while the "o" format is invariant for **DateOnly**.

**Comment by meziantou** [comment]
I tried to share more logic between MA0076 and MA0011. So, both rules should report less false-positives.
